server/jetstream_events.go

```diff
@@ -295,3 +295,15 @@ type JSServerRemovedAdvisory struct {
 	Cluster  string `json:"cluster"`
 	Domain   string `json:"domain,omitempty"`
 }
+
+// JSAPIRateLimitAdvisoryType is sent when the JS API request queue limit is reached.
+const JSAPIRateLimitAdvisoryType = "io.nats.jetstream.advisory.v1.api_rate_limit"
+
+// JSAPIRateLimitAdvisory is a advisory published when JetStream limits API access due to rate limits
+type JSAPIRateLimitAdvisory struct {
+	TypedEvent
+	Server      string `json:"server"`            // Server that created the event, name or ID
+	Domain      string `json:"domain,omitempty"`  // Domain the server belongs to
+	Account     string `json:"account,omitempty"` // Optionally account that hit the limit, or empty for system
```

Comment:
Will be a hard limit in 2.11 even with flow control, so can remove.

Reply:
We discussed that we want to introduce a limit for like a free account can have 10 req/sec or something, so there will be a concept of an account limit and that's why I added this here now so that if we release CLI before 2.11 when that comes it'll already know how to handle these messages and report the account

Reply:
ok but them maybe this is the wrong advisory here. This one is hard limit, and we drop everything. RateLimit we will most likely drop last request in hand and probably respond to the app tier (with possible delay etc).

So I think this one should be different for its purpose.

Reply:
yeah that's fair.

So we can drop `account` here and later introduce a new advisory on new subjects etc

Reply:
I would name advisory different as well. Not really a rate limit here, LimitReached?

Reply:
Yeah